File: src/main/java/com/travel/domain/travel/domain/Travel.java
package com.travel.domain.travel.domain;

import com.travel.domain.travel.domain.city.TravelCity;
import com.travel.domain.travel.domain.plan.TravelPlan;
import com.travel.domain.travel.dto.TravelUpdateDto;
import com.travel.global.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Getter
@NoArgsConstructor
@Entity
@Table(name="travel")
public class Travel extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(mappedBy = "travel", cascade = CascadeType.ALL, orphanRemoval = true)
    private final List<TravelPlan> travelPlanList = new ArrayList<>();

    @OneToMany(mappedBy = "travel", cascade = CascadeType.ALL, orphanRemoval = true)
    private final List<TravelCity> travelCityList = new ArrayList<>();

    @Column
    private String name;

    @Column
    private String userId;

    @Column
    private String travelStartDate;

    @Column
    private String travelEndDate;

    @Builder
    public Travel(Long id, String name, String userId, String travelStartDate, String travelEndDate) {
        this.id = id;
        this.name = name;
        this.userId = userId;
        this.travelStartDate = travelStartDate;
        this.travelEndDate = travelEndDate;
    }

    public void update(TravelUpdateDto travelUpdateDto) {
        this.name = travelUpdateDto.getName();
        this.travelStartDate = travelUpdateDto.getTravelStartDate();
        this.travelEndDate = travelUpdateDto.getTravelEndDate();
    }
}

File: src/main/kotlin/com/travel/domain/travel/domain/Travel.kt
package com.travel.domain.travel.domain

import com.travel.domain.travel.domain.city.TravelCity
import com.travel.domain.travel.domain.plan.TravelPlan
import com.travel.domain.travel.dto.JavaTravelUpdateDto
import com.travel.global.base.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "travel")
class Travel(
        @Column
        var name: String,

        @Column
        val userId: String,

        @Column
        var travelStartDate: String,

        @Column
        var travelEndDate: String,

        @OneToMany(mappedBy = "travel", cascade = [CascadeType.ALL], orphanRemoval = true)
        val travelPlanList: List<TravelPlan> = mutableListOf(),

        @OneToMany(mappedBy = "travel", cascade = [CascadeType.ALL], orphanRemoval = true)
        val travelCityList: List<TravelCity> = mutableListOf(),

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null
): BaseEntity() {
        fun update(travelUpdateDto: JavaTravelUpdateDto) {
                this.name = travelUpdateDto.name
                this.travelStartDate = travelUpdateDto.travelStartDate
                this.travelEndDate = travelUpdateDto.travelEndDate
        }
}